### PR TITLE
travis: Push converted tags to ocean-d2 repo

### PR DESCRIPTION
**Pull request by llucax**

### CODE REVIEW

**Comment by oraby** [comment]
LGTM, I've tested it on a local repo and it works as expected.
Let's wait for a bit to see if we should include dub information as well or we'd leave it for a later time.

On unrelated note, I was reading about [SemVer](http://semver.org/) to see why you used `+d2.auto` and it turned out that you are adding metdata, so all is good. However, it mentions that  the hyphen character  should  be added for pre-releases. but we use it for `-preview` which is actually more of "end-of-life" support. Made we should use `+preview` in the future instead.

**Comment by oraby** [comment]
> On unrelated note, I was reading about SemVer to see why you used +d2.auto and it turned out that you are adding metdata, so all is good. However, it mentions that the hyphen character should be added for pre-releases. but we use it for -preview which is actually more of "end-of-life" support. Made we should use +preview in the future instead.

Oh crap, I should read your message to the end in the future :p, you've just mentioned something similar in #84

**Comment by oraby** [comment]
An extra thought: if we want to add dub and modified README then one way to do it is still to add a a sync branch in the d2 repo and we can change the `after-success.sh` to basically download the latest version of convert code as in [here](https://github.com/oraby/ocean/blob/727d6eea53ccb2fbb16dc8eea44f16a993ca528c/sync_d2_start.sh) for example.
If we'd do that then we'd add the sync branch commit hash in the commit message.

**Comment by llucax** [comment]
> However, it mentions that the hyphen character should be added for pre-releases. but we use it for -preview which is actually more of "end-of-life" support. Made we should use +preview in the future instead.

`-preview` was only done when this wasn't the real upstream repo for the main development. The tag was fine because the idea was to mark this as a pre-release, since nobody was actually using these tags. Now that the development was moved here, we should completely forget about `-preview` tags.

**Comment by llucax** [comment]
> An extra thought: if we want to add dub and modified README then one way to do it is still to add a a sync branch in the d2 repo and we can change the after-success.sh to basically download the latest version of convert code as in here for example.

For me this is too much complexity. If we want to add a dub file (I wouldn't add a readme file), we can just include it as part of the script.

**Comment by llucax** [comment]
Updated to use a more complex but secure method for authentication (if the user/password are included in the URL, any errors or messages from Git will reveal the credentials).

**Comment by llucax** [comment]
I did a quick try and it seems to work, we'll just need to test the credentials + travis integration in practice though.

**Comment by oraby** [comment]
Here's my go at `dub.json`:
```json
{
    "name": "ocean-d2",
    "description": "Auto convert of ocean-d1 library. Ocean is general purpose library with a focus on supporting the development of high-performance, real-time applications.",
    "authors": ["Sociomantic Labs GmbH "],
    "homepage": "https://github.com/sociomantic-tsunami/ocean",
    "license": "BSD 3-clause or Check Ocean License",
    "copyright": "Copyright (c) 2004-2017 Sociomantic Labs GmbH and Tango contributors.\nAll rights reserved.",
    "systemDependencies": "Depending on the component used you might need one of the following libraries:\n-lglib-2.0 -lpcre -lxml2 -lxslt -lebtree -lreadline -lhistory -llzo2 -lbz2 -lz -ldl -lgcrypt -lgpg-error -lrt.\nNotice that the used -lebtree comes from https://github.com/sociomantic-tsunami/ebtree",
    "importPaths": ["src/"],
    "targetType": "sourceLibrary"
}

```

**Comment by llucax** [comment]
> "name": "ocean-d2",

Any reason not to use just "ocean"?

> "homepage": "https://github.com/sociomantic-tsunami/ocean",

And here use ocean-d2, as it is from where the code really comes?

**Comment by oraby** [comment]
> Any reason not to use just "ocean"?

Basically I want to reserve the `ocean` name for when we move this (ocean-d1) repo to be full d2 repo.

> "homepage": "https://github.com/sociomantic-tsunami/ocean",

As far as I understand, that needs to be the project homepage. That can be the read-the-docs or something. So I thought to add the mian-repo as the homepage as it's where all the development happens.

**Comment by llucax** [comment]
> Basically I want to reserve the ocean name for when we move this (ocean-d1) repo to be full d2 repo.

👍 

> As far as I understand, that needs to be the project homepage. That can be the read-the-docs or something. So I thought to add the mian-repo as the homepage as it's where all the development happens.

I think this might be a bit confusing, it should be clearer to the user to send them to the page that explains they are using an automated conversion of a project, not really the project itself.

**Comment by llucax** [comment]
I just pushed an update adding the dub file, I converted it to `.sdl` and changed the URL, but we can keep discussing the URL :)

File: ci/ocean-d2-dub.sdl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name "ocean-d2"
+description "Auto convert of ocean-d1 library. Ocean is general purpose library with a focus on supporting the development of high-performance, real-time applications."
+authors "Sociomantic Labs GmbH "
```
Comment (oraby) [inline ci/ocean-d2-dub.sdl]: extra space after GmbH

**Comment by llucax** [comment]
Pushed new updated, I reverted the URL because DUB already shows the repo URL, so putting the original project seems reasonable.

**Comment by llucax** [comment]
OK, we already sort of pushed this online for Hatem's lighting talk, so I will merge. Let's see how it works next time we tag something.

FYI @gavin-norman-sociomantic @mihails-strasuns-sociomantic

**Pull request merged**